submit.py

```diff
@@ -1,15 +1,27 @@
 #!/usr/bin/env python
-import ConfigParser
+from __future__ import print_function
 import optparse
 import os
 import sys
 import itertools
-import mimetools
 import mimetypes
-import urllib
-import urllib2
 import traceback
-
+import random
+import string
+if sys.version_info[0] >= 3:
+    _PYTHON_VERSION = 3
+    import configparser
+    import urllib.parse
+    import urllib.request
+    import urllib.error
+else:
+    # Python 2, import modules with Python 3 names
+    _PYTHON_VERSION = 2
+    import ConfigParser as configparser
+    import urllib
+    import urllib2
+    urllib.request = urllib.error = urllib2
```

Comment:
Ok. Merge. I still think ugly is better than deceptive but it can be fixed
later. Also update in Kattis repo?

On Friday, 15 January 2016, austrin <notifications@github.com> wrote:

> In submit.py
> <https://github.com/Kattis/kattis-cli/pull/3#discussion_r49832480>:
>
> > -
> > +import random
> > +import string
> > +if sys.version_info[0] >= 3:
> > +    _PYTHON_VERSION = 3
> > +    import configparser
> > +    import urllib.parse
> > +    import urllib.request
> > +    import urllib.error
> > +else:
> > +    # Python 2, import modules with Python 3 names
> > +    _PYTHON_VERSION = 2
> > +    import ConfigParser as configparser
> > +    import urllib
> > +    import urllib2
> > +    urllib.request = urllib.error = urllib2
>
> @mikaelgoldmann <https://github.com/mikaelgoldmann> suggestion: merge
> this as is for now since it works and is tested, then we can discuss a bit
> more and if desired easily change according to your suggestion separately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Kattis/kattis-cli/pull/3/files#r49832480>.
>